lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -285,37 +296,107 @@ class BitmapCanvas extends EngineCanvas {
     _canvasPool.clipPath(path);
   }
 
+  /// Whether drawing operation should use DOM node instead of Canvas.
+  ///
+  /// - Perspective transforms are not supported by canvas and require
+  ///   DOM to render correctly.
+  /// - Pictures typically have large rect/rounded rectangles as background
+  ///   prefer DOM if canvas has not been allocated yet.
+  bool _useDomForRendering(SurfacePaintData paint) =>
+      _contains3dTransform || (_canvasPool._canvas == null
+          && paint.maskFilter == null && paint.shader == null &&
+          paint.style != ui.PaintingStyle.stroke);
+
   @override
   void drawColor(ui.Color color, ui.BlendMode blendMode) {
-    _canvasPool.drawColor(color, blendMode);
+    final SurfacePaintData paintData = SurfacePaintData()
+      ..color = color
+      ..blendMode = blendMode;
+    if (_useDomForRendering(paintData)) {
+      drawRect(ui.Rect.fromLTWH(0, 0, _bounds.width, _bounds.height),
```

Comment:
Will this cover the entire picture in the presence of transforms? In the `CanvasPool` we do `ctx.fillRect(-10000, -10000, 20000, 20000)` to emulate `drawColor`.

Reply:
Done.